test/external/modules/notify/controller.js

```diff
@@ -0,0 +1,68 @@
+const { expect } = require('@hapi/code');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+const { experiment, test, beforeEach, afterEach, before } = exports.lab = require('@hapi/lab').script();
+
+const services = require('external/lib/connectors/services');
+const controller = require('external/modules/notify/controller');
+
+experiment('callback', () => {
+  let request = {
+    method: 'POST',
+    url: '/notify/callback',
+    headers: { authorization: `Bearer test` },
+    payload: {}
+  };
+
+  const h = {
+    response: sinon.stub().returns({
+      code: sinon.spy()
+    }),
+    code: sinon.stub().resolves()
+  };
+
+  beforeEach(async () => {
+    sandbox.stub(services.water.notify, 'postNotifyCallback').resolves();
+  });
+
+  afterEach(async () => {
+    sandbox.restore();
+  });
+
+  experiment('when it has no token', () => {
+    before(async () => {
+      request.headers = {};
+    });
+    test('returns Unauthorized', async () => {
+      await controller.callback(request, h);
+      expect(h.response.calledWith('Unauthorized')).to.be.true();
+      expect(h.response.lastCall.returnValue.code.lastCall.args).to.equal([403]);
```

Comment:
This is fine as is, but setting up your stubs slightly differently, could make this a lot easier to read:
```
const responseStub = {
      code: sandbox.stub()
    };

h = {
    response: sinon.stub().returns(responseStub),
   ... }
```

Your test can then be writted as `expect(responseStub.code.calledWith(403)).to.be.true();`

Reply:
Cool. That seems tidier so I'll go with that :)